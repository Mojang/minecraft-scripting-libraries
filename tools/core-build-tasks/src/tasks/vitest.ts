import { execSync } from 'child_process';

export function vitestTask() {
    return () => {
        execSync('vitest', { stdio: 'inherit' });
    };
}
