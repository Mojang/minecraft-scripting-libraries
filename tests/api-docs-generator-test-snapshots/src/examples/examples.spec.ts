// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

import { describe, it } from 'vitest';
import { runGeneratorForTest } from '../runGeneratorForTest';

describe('Examples', () => {
    it('Properly generates examples documentation and definitions', () => {
        runGeneratorForTest({
            testDir: __dirname,
            generators: ['msdocs'],
        });
    });
});
