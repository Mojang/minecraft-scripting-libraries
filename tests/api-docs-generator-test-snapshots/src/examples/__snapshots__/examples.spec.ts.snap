// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Examples > Properly generates examples documentation and definitions > # Of Files Generated 1`] = `4`;

exports[`Examples > Properly generates examples documentation and definitions > msdocs/scriptapi/TOC.yml 1`] = `
"  - name: generated-module
    items:
    - name: generated-module
      href: generated-module/generated-module.md
    - name: generated-module Changelog
      href: generated-module/changelog.md
    - name: ScriptGeneratedClass
      href: generated-module/ScriptGeneratedClass.md
"
`;

exports[`Examples > Properly generates examples documentation and definitions > msdocs/scriptapi/generated-module/ScriptGeneratedClass.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: generated-module.ScriptGeneratedClass Class
description: Contents of the generated-module.ScriptGeneratedClass class.
---
# ScriptGeneratedClass Class

A script generated class

## Properties

### **testProperty**
\`testProperty: string | undefined;\`

A script generated property

Type: *string | undefined*

Notes:
  - This property can't be used in read-only mode.

## Methods
- [testFunctionWithTSDocComments](#testfunctionwithtsdoccomments)

### **testFunctionWithTSDocComments**
\`
testFunctionWithTSDocComments(tsdocParam: number, msdocParam: number): boolean
\`

A script generated function with doc text defined in TSDoc comments

#### **Parameters**
- **tsdocParam**: *number*
  
  Param comment from TSDoc
- **msdocParam**: *number*
  
  This overwrites the TSDoc for msdocParam with an info.json file's description

**Returns** *boolean* - This is a return comment defined in info.json
  
Notes:
- This function can't be called in read-only mode.
- This function can throw errors.
  - Throws comment in TSDocs

#### Examples

##### ***addTranslatedSign.ts***

\`\`\`typescript
import { DimensionLocation, world, BlockPermutation, BlockComponentTypes } from '@minecraft/server';

function placeTranslatedSign(location: DimensionLocation, text: string) {
    const signBlock = location.dimension.getBlock(location);

    if (!signBlock) {
        console.warn('Could not find a block at specified location.');
        return;
    }
    const signPerm = BlockPermutation.resolve('minecraft:standing_sign', { ground_sign_direction: 8 });
    signBlock.setPermutation(signPerm);

    const signComponent = signBlock.getComponent(BlockComponentTypes.Sign);
    if (signComponent) {
        signComponent.setText({ translate: 'item.skull.player.name', with: [text] });
    } else {
        console.error('Could not find a sign component on the block.');
    }
}

placeTranslatedSign(
    {
        dimension: world.getDimension('overworld'),
        x: 0,
        y: 0,
        z: 0,
    },
    'Steve'
);
\`\`\`

(preview) Work with this sample on the [MCTools.dev](https://mctools.dev/?open=gp/addTranslatedSign.ts) code sandbox.
"
`;

exports[`Examples > Properly generates examples documentation and definitions > msdocs/scriptapi/generated-module/changelog.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: generated-module Changelog
description: Changelog of the \`generated-module\` module
---
# \`generated-module\` Changelog

## Version Changes
- [1.0.0-beta](#100-beta)

## 1.0.0-beta
#### Added \`generated-module\` Module
"
`;

exports[`Examples > Properly generates examples documentation and definitions > msdocs/scriptapi/generated-module/generated-module.md 1`] = `
"---
# DO NOT TOUCH — This file was automatically generated by @minecraft/api-docs-generator, to report problems file an issue at https://github.com/Mojang/minecraft-scripting-libraries
author: jakeshirley
ms.author: jashir
ms.service: minecraft-bedrock-edition
ms.date: 02/10/2025
title: generated-module Module
description: Contents of the generated-module module
monikerRange: "=minecraft-bedrock-experimental"
---
# \`generated-module\` Module

> [!CAUTION]
> This module is still in pre-release.  It may change or it may be removed in future releases.

## [Changelog](changelog.md)

## Manifest Details
\`\`\`json
{
    "module_name": "generated-module",
    "version": "1.0.0-beta"
}
\`\`\`
This is version 1.x.x of this module, which is the latest as of version 1.0.0 of Minecraft.

## Available Versions
- \`1.0.0-beta\`

## Classes
- [ScriptGeneratedClass](ScriptGeneratedClass.md)
"
`;
